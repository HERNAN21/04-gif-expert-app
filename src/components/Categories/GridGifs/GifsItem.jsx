import PropTypes from 'prop-types'
export const GifsItem = ({title,url,id}) => {

    // console.log(title,url,id);

    return (
        <>
            <div className="card">
                <img src={url} alt={title} />
                <p>{title}</p>
            </div>
        </>
    )
}

GifsItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    
}
