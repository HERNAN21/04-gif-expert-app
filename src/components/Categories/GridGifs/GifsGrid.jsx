import { useFetchGifs } from '../../../hooks/useFetchGifs';
import { GifsItem } from './GifsItem';

export const GifsGrid = ({category}) => {

  const {images,isLoading} = useFetchGifs(category);

  console.log(isLoading);


  return (
    <>
      <div>
        <h3>{category}</h3>
        {isLoading && (<h2>Cargando...</h2>)}
        <div className="card-grid">
          {
              images.map((image) =>(
                  <GifsItem key={image.id} {...image}/>
              ))
          }
        </div>
      </div>
    </>
  )
}
