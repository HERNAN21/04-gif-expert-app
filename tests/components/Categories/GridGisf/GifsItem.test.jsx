
import { render,screen } from "@testing-library/react"
import { GifsItem } from "../../../../src/components/Categories/GridGifs/GifsItem"

describe('Puebas en <GifsItem/>', () => { 
    const title = "asd";
    const url = "http://";

    test('debe hacer match con el snapshot', () => { 
        const {container}=render(<GifsItem title={title} url={url} />)

        expect(container).toMatchSnapshot();

    })

    test('Debe mostrar la imagen con el url Indicado', () => {
        render(<GifsItem title={title} url={url}/>)
        //   screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })
    
    test('debe de mostrar el titulo en el componente ', () => { 
        render(<GifsItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
     })

 })