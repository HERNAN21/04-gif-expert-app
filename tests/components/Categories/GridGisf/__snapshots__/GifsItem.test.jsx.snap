// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <GifsItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="asd"
      src="http://"
    />
    <p>
      asd
    </p>
  </div>
</div>
`;
